CMakeLists.txt

```diff
@@ -164,10 +164,30 @@ add_subdirectory( core )
 add_subdirectory( device/common )
 if( TRACCC_BUILD_CUDA )
    add_subdirectory( device/cuda )
+else()
+   if( TRACCC_BUILD_KOKKOS )
+      add_subdirectory( device/cuda )
+   endif()
 endif()
 if( TRACCC_BUILD_SYCL )
    add_subdirectory( device/sycl )
 endif()
+if( TRACCC_BUILD_KOKKOS )
+   # Set up Kokkos.
+   option( TRACCC_SETUP_KOKKOS
+     "Set up the Kokkos library" TRUE )
+   option( TRACCC_USE_SYSTEM_KOKKOS 
+      "Pick up an existing installation of Acts from the build environment"
+      TRUE )
+   if( TRACCC_SETUP_ACTS )
+      if( TRACCC_USE_SYSTEM_KOKKOS )
+         find_package( Kokkos REQUIRED )
+      else()
+         add_subdirectory( extern / kokkos )
+      endif() 
+   endif() 
+   add_subdirectory( device/kokkos )
+endif()
```

Comment:
You're mixing the setup of Kokkos itself, and the inclusion of your Kokkos library in this place.

In order to set up Kokkos, just put:

```cmake
option( TRACCC_SETUP_KOKKOS
  "Set up the Kokkos library" TRUE )
if( TRACCC_SETUP_KOKKOS )
   find_package( Kokkos REQUIRED )
endif()
```

Now... I would actually love it if Kokkos could be built as part of this project, using code in an `extern/kokkos` subdirectory. But you didn't add that in this PR.

In any case, in this location in the code you should just have:

```cmake
if( TRACCC_BUILD_KOKKOS )
   add_subdirectory( device/kokkos )
endif()
```

While not forgetting to add

```cmake
option( TRACCC_BUILD_KOKKOS "Build the Kokkos sources included in traccc" FALSE )
```

a bit earlier in this file as well.